docs/references/architecture/adr-113-mempool-lanes.md

```diff
@@ -3,43 +3,132 @@
+disseminated in the same order in which the mempool has received them.
+
+More formally, given any two different transactions `tx1` and `tx2`, if the mempool receives `tx1`
+before receiving `tx2`, then:
+- `tx1` will be validated against the application (via `CheckTx`) before `tx2`, and
+- `tx1` will be processed and disseminated before `tx2`.
+
+Note that a node's mempool can receive a transaction either from a `broadcast_tx_*` RPC endpoint or
+from a peer.
+
+### Mempool with QoS
+
+With the introduction of *transaction classes*, the main goal is to guarantee that certain
+transactions have lower latency than others. 
+
+:memo: _Definition_: a *transaction class* is a disjoint set of transactions having some common
+characteristics as defined by the application.
+
+A transaction may only have one class. If it is not assigned any specific class, it will be assigned
+a *default class*, which is a special class always present in any set of classes. Because no
+transaction can belong to two or more classes, transaction classes form a disjoint set. Also, all
+transactions in the mempool are the union of the transactions in all classes.
+
+:memo: _Definition_: Each class has a *priority* and two classes cannot have the same priority. Then
+all classes can be ordered by priority.
 
-Try using "QoS" term. Because that's what it actually is.
+Now, given these definitions, we want the proposed QoS mechanism to offer the following property:
 
-We need problem definition.
+:parking: _Property_ **Priorities between classes**: Transactions belonging to a certain class will
+be processed and disseminated before transactions belonging to another class with lower priority.
 
+Formally, given two transaction classes `c1` and `c2`, with `c1` having more priority than `c2`, if
+the application assigns via `CheckTx` the classes `c1` and `c2` respectively to transactions `tx1`
+and `tx2`, then `tx1` will be processed and disseminated before `tx2`.
 
-Definitions
-Tx1, Tx2
-Traffic classes
-Traffic class order (partial?)
-Latency(tx) --- important
+More importantly, as a direct consequence of this property, `tx1` will have a lower latency than
+`tx2`, because `tx1` will be disseminated faster and it will be included in a block before `tx2`.
+Currently, it is not possible to guarantee this kind of property.
 
-Properties / Requirements (similar to abci++ requirement)
+:memo: _Definition_: The *latency of a transaction* is the difference between the time at which the
+transaction was received for the first time by a node, and the timestamp of the block in which the
+transaction finally was included.
 
-[Examples]
-* Property 1: tx1 --> tc1, tx2 --> tc1 ---> FIFO
+We want also to keep the FIFO ordering within each class:
 
-* Property 2: tx1 --> tc1, tx2 --> tc2; tc1 < tc2 --> tx1 first, then tx2
+:parking: _Property_ **Sequential ordering per class**: For transactions within the same class, the
```

Comment:
Nit. Reuse the same terminology, 'FIFO', as above. Or am I missing something?

Reply:
That's right. I changed it to FIFO.